### PR TITLE
[SRVKS-610] Update Kourier to 0.17

### PR DESCRIPTION
**Pull request by nak3**
This patch updates Kourier to 0.17.

This patch solved SRVKS-610 by `sed` command as:

```
sed -i -e 's/kourier-control.knative-serving/kourier-control.knative-serving-ingress/g'  kourier-${KOURIER_VERSION}.yaml
```

but we should replace it with operator.

/cc @markusthoemmes

### CODE REVIEW

**Comment by nak3** [comment]
/hold

I think that we don't need following change as gateway and controller is in the same namespace.

```
          hosts:
            - socket_address:
                address: "kourier-control.knative-serving-ingress"
```

I will update this.

**Comment by nak3** [comment]
I wanted to replace [`address: "kourier-control.knative-serving"`](https://github.com/knative-sandbox/net-kourier/blob/release-0.17/deploy/kourier-knative.yaml#L280) with `address: "kourier-control.knative-serving-ingress"` by operator but it seems difficult. I used `sed` as a temporary solution.

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

/hold

To wait for all tests to pass.

File: knative-operator/deploy/resources/kourier/kourier-release-0.17.yaml [hunk 1]
```diff
@@ -6,73 +6,6 @@ metadata:
     networking.knative.dev/ingress-provider: kourier
 ---
 apiVersion: v1
-kind: ConfigMap
-metadata:
```
Comment (matzew) [inline knative-operator/deploy/resources/kourier/kourier-release-0.17.yaml:10]: @aliok look, Ali!  Removal of things we like to remove as well :joy_cat:

Comment (aliok) [reply knative-operator/deploy/resources/kourier/kourier-release-0.17.yaml:10]: LOL :laughing: 

I think my idea of having a unit test that checks the manifest is useful in this case too.

These kind removals can be forgotten (happened to us)

**Comment by nak3** [comment]
/retest

```
I1013 07:23:14.540445   53132 request.go:621] Throttling request took 11.244896936s, request: GET:https://api.ci-op-mbsmcmd2-85c73.origin-ci-int-aws.dev.rhcloud.com:6443/apis/discovery.k8s.io/v1beta1?timeout=32s
make: *** [test-upstream-e2e-no-upgrade] Error 9
error: failed to execute wrapped command: exit status 2
```

Not sure why it failed. At least Kourier looks working fine.

**Pull request merged**